### ISSUE TITLE
RPM build for calico latest stable version

### ISSUE THREAD

**Issue by dpavlos** [mantl/mantl#1941]
Currently we use this [version](https://bintray.com/shippedrepos/mantl-1.3/calico) which is pretty old. We can keep both versions for the time being (if possible). https://github.com/mantl/mantl/issues/1928

**Comment by Theaxiom** [comment]
@stevendborrelli @Zogg Any words of wisdom regarding this process? 👍

**Comment by Theaxiom** [comment]
Here's the packaging spec for Calico in Mantl: https://github.com/mantl/mantl-packaging/blob/4961400efbc881bab355e84bf7d6df47d691a16e/calico/calico/spec.yml

**Comment by Theaxiom** [comment]
They've deprecated calico-mesos:

Note: Mesos has deprecated support for the Net-Modules networking interface in favor of the standardized Container Networking Interface (CNI).

Per: https://github.com/projectcalico/netmodules-plugin

**Comment by Theaxiom** [comment]
Okay, attempting to package latest version: https://travis-ci.org/mantl/mantl-packaging/builds/225029503

**Comment by Theaxiom** [comment]
Alright folks, here ya go, see if it works: https://bintray.com/mantl/rpm-packages/calico/1.1.1-1

You'll have to change the `calico_version` in `roles/calico/defaults/main.yml` to version `1.1.1`

**Comment by Theaxiom** [comment]
When deploying Mantl with the latest version of Calico, I receive this error:

```
TASK: [calico | add calico ip-ip pool if required] ****************************
failed: [control-01] => {"changed": true, "cmd": "calicoctl pool add \"192.168.0.0/16\" --ipip --nat-outgoing", "delta": "0:00:00.119116", "end": "2017-04-24 00:36:50.870788", "rc": 1, "start": "2017-04-24 00:36:50.751672", "warnings": []}
stderr: Unknown command: "pool"
stdout: Usage:
  calicoctl [options] <command> [<args>...]

    create    Create a resource by filename or stdin.
    replace   Replace a resource by filename or stdin.
    apply     Apply a resource by filename or stdin.  This creates a resource
              if it does not exist, and replaces a resource if it does exists.
    delete    Delete a resource identified by file, stdin or resource type and
              name.
    get       Get a resource identified by file, stdin or resource type and
              name.
    config    Manage system-wide and low-level node configuration options.
    ipam      IP address management.
    node      Calico node management.
    version   Display the version of calicoctl.

Options:
  -h --help               Show this screen.
  -l --log-level=<level>  Set the log level (one of panic, fatal, error,
                          warn, info, debug) [default: panic]

Description:
  The calicoctl command line tool is used to manage Calico network and security
  policy, to view and manage endpoint configuration, and to manage a Calico
  node instance.

  See 'calicoctl <command> --help' to read about a specific subcommand.

FATAL: all hosts have already failed -- aborting

PLAY RECAP ********************************************************************
common | install system utilities -------------------------------------- 77.79s
consul | enable nginx-consul ------------------------------------------- 68.63s
docker | install docker packages --------------------------------------- 52.32s
docker | enable docker ------------------------------------------------- 28.75s
calico | install calico ------------------------------------------------ 24.13s
vault | enable vault --------------------------------------------------- 20.41s
consul | install consul ------------------------------------------------ 18.64s
calico | ensure calico docker image is present ------------------------- 14.55s
mesos | wait for zookeeper service to be registered -------------------- 13.15s
docker | install latest device-mapper-libs ----------------------------- 11.02s
           to retry, use: --limit @/home/vagrant/sample.retry

control-01                 : ok=193  changed=155  unreachable=0    failed=1
edge-001                   : ok=115  changed=99   unreachable=0    failed=0
kubeworker-001             : ok=115  changed=99   unreachable=0    failed=0
localhost                  : ok=0    changed=0    unreachable=0    failed=0
worker-001                 : ok=115  changed=99   unreachable=0    failed=0

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
mobility:mantl trowland$
```

**Comment by thomasvincent** [comment]
https://github.com/mantl/mantl/blob/master/roles/calico/tasks/calico-configure.yml <https://github.com/mantl/mantl/blob/master/roles/calico/tasks/calico-configure.yml>

**Comment by Theaxiom** [comment]
Yeah, obviously the API has changed. Trying to figure out the new nomenclature right now.

**Comment by thomasvincent** [comment]
http://docs.projectcalico.org/v1.5/reference/calicoctl/pool <http://docs.projectcalico.org/v1.5/reference/calicoctl/pool>

**Comment by Theaxiom** [comment]
http://docs.projectcalico.org/v1.5/reference/calicoctl/

**Comment by Theaxiom** [comment]
Hah, thanks @thomasvincent :)

**Comment by Theaxiom** [comment]
Here's the most recent documentation for Calico: http://docs.projectcalico.org/v2.1/reference/calicoctl/commands/

**Comment by Theaxiom** [comment]
It looks like the API has changed for sure.

**Comment by Theaxiom** [comment]
I believe this is related to the deprecation, see these repos:
- https://github.com/containernetworking/cni
- https://github.com/projectcalico/cni-plugin

**Comment by Theaxiom** [comment]
They are now using `CNI - the Container Network Interface`

**Comment by Theaxiom** [comment]
I joined the Calico Users on Slack and I was informed of the following:

You can get a listing of the pools with `calicoctl get pools -o yaml`  or create or modify a pool with a `calicoctl apply -f pool.yaml` type command.  See http://docs.projectcalico.org/v2.1/reference/calicoctl/resources/ippool for the configuration parameters.

**Comment by dpavlos** [comment]
We will probably have to modify the profile creation too. https://github.com/mantl/mantl/blob/master/roles/calico/tasks/calico-configure.yml#L24

**Comment by Theaxiom** [comment]
You're probably correct.

**Comment by KaGeN101** [comment]
I have done this in the role busy testing

**Comment by Theaxiom** [comment]
The profile can probably be created inside the same YAML file.

**Comment by KaGeN101** [comment]
Okay that YAML change applies another commad that does not work

failed: [mantl-dev1-control-01] => {"changed": false, "cmd": "calicoctl profile show|grep -w \"dev\"", "delta": "0:00:00.070784", "end": "2017-04-24 09:50:07.951813", "failed": true, "failed_when_result": true, "rc": 1, "start": "2017-04-24 09:50:07.881029", "stdout_lines": [], "warnings": []}
stderr: Unknown command: "profile"

**Comment by Theaxiom** [comment]
Yeah, that's what @dpavlos were just discussing.

**Comment by Theaxiom** [comment]
http://docs.projectcalico.org/v2.1/reference/calicoctl/resources/profile

**Comment by KaGeN101** [comment]
Yeah look like its separate YAML adding now

**Comment by Theaxiom** [comment]
Try this:

```
apiVersion: v1
kind: profile
metadata:
  name: {{ calico_profile }}
```

**Comment by KaGeN101** [comment]
Do we only need the name or the spec as well?

```
apiVersion: v1
kind: profile
metadata:
  name: profile1
  labels:
    profile: profile1 
spec:
  ingress:
  - action: deny
    source:
      net: 10.0.20.0/24
  - action: allow
    source:
      selector: profile == 'profile1'
  egress:
  - action: allow 
```

**Comment by Theaxiom** [comment]
Previously it was only passing name.

**Comment by Theaxiom** [comment]
If that doesn't work, try this:
```
apiVersion: v1
kind: profile
metadata:
  name: {{ calico_profile }}
  labels:
    profile: {{ calico_profile }} 
```

**Comment by Theaxiom** [comment]
Ultimately, I think it would be a wise decision to set a proper route table here.

**Comment by Theaxiom** [comment]
I just confirmed with Calico devs that spec is not required and default is allow all. They are going to update the docs to reflect that.

**Comment by Theaxiom** [comment]
https://github.com/projectcalico/calico/issues/709

**Comment by KaGeN101** [comment]
Get this for node status this looks like it is the k8 host network pods
Cant see anything for 192.168.0.0/16

This is about the sum total of things I can find the calico pods are just stuck in ContianerCreating. I presume they have no access to the internet hence they can't pull the image

```
sudo calicoctl node status
Calico process is running.

IPv4 BGP status
+--------------+-------------------+-------+----------+-------------+
| PEER ADDRESS |     PEER TYPE     | STATE |  SINCE   |    INFO     |
+--------------+-------------------+-------+----------+-------------+
| 10.1.1.203   | node-to-node mesh | up    | 14:23:23 | Established |
| 10.1.2.231   | node-to-node mesh | up    | 14:23:24 | Established |
| 10.1.3.206   | node-to-node mesh | up    | 14:23:24 | Established |
| 10.1.2.43    | node-to-node mesh | up    | 14:23:23 | Established |
| 10.1.1.50    | node-to-node mesh | up    | 14:23:24 | Established |
| 10.1.2.153   | node-to-node mesh | up    | 14:23:23 | Established |
+--------------+-------------------+-------+----------+-------------+

IPv6 BGP status
No IPv6 peers found
```

**Comment by dpavlos** [comment]
@KaGeN101 Same here. I can't see any workload endpoints and pods are stucked on "creating" state. Seems that we are missing something with the cni plugin configuration.

kubelet service on kubeworker gives the following error. 

```
# service kubelet status -l
[...]
kubeworker-001 kubelet[16755]: E0424 16755 cni.go:123] Error while adding to cni network: invalid configuration: no configuration has been provided
```

**Comment by KaGeN101** [comment]
Will have to check the that CNI is configured when configuring k8 I guess. Was probably not used before but is now will dig around there

**Comment by KaGeN101** [comment]
I get this error now on latest calico-upgrade branch:
failed: [mantl-dev1-control-01] => {"elapsed": 90, "failed": true}
msg: Timeout when waiting for 127.0.0.1:8085

FATAL: all hosts have already failed -- aborting

**Comment by KaGeN101** [comment]
Still timeouts here 
TASK: [kubernetes-master | wait for apiserver to come up] ********************* 
:/

**Comment by dpavlos** [comment]
Fixed in master #1963. Closing the issue.
